go.mod

```diff
@@ -17,7 +16,7 @@ require (
 	github.com/graphql-go/handler v0.2.3
 	github.com/magefile/mage v1.14.0
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/moby/buildkit v0.10.5
+	github.com/moby/buildkit v0.10.1-0.20221026024300-46f6f51822ff
```

Comment:
This is going to break the automatic spin up of buildkitd.

I'm personally okay with moving to being based on master branch builds of buildkit, but making that switch now is going to increase the scope here of this PR. Maybe it's doable relatively quickly, but if not I don't think we should block the PR on it. We can use the copied source for now and then switch to master buildkit in a follow up.

Reply:
Oh right - I just had to fix that code to handle `replace` rules, so I think I can just tweak it again to use `master` if the version isn't a regular semver.

Reply:
The problem is that the code looks at the version of buildkit in our dependencies and then pulls the buildkit image based on that: https://github.com/sipsma/dagger/blob/ab7fba6ce6611a5eb0c279f8db42bb45b1cca11f/internal/buildkitd/buildkitd.go#L85-L85

So before it was v0.10.5 and we thus pulled `moby/buildkit:v0.10.5`.

There are images built from the master branch but I don't know of a way to map the commit SHA we depend on to those images. Honestly, I think I'd be okay with just hardcoding an image digest to pull for moby/buildkit and then adding a test to make sure we always keep it updated with our client. Not having to wait for official buildkit releases will resolve so many headaches (i.e. the dumb issue where running `go get -u` breaks the fsutil dep or [this fix](https://github.com/tonistiigi/fsutil/commit/e77355bad25d4deb717516096cf3cd216c7dc979) from May that was made for a bug encountered by a Dagger Europa user which still has not made it to us...)

I think we can then add a test that makes sure our client dependency is aligned with the server; when I currently run `docker exec -it dagger-buildkitd buildctl --version` I get a commit SHA, so I think we can use that in the test (since buildctl version inside that container should be the same as buildkitd).

*THEN* when we switch to having our own embedded buildkitd and image and all that I think this will likely all still be useful and what we want, so probably not throwaway work either.

Reply:
Again, this might be too much for this PR, another option is to go back to copying code for a bit, then make these changes in a follow up. I'm also happy to do that follow up too to help distribute the workload here.

Reply:
I think I came up with something that should cover all of the following cases:

* Pointing to an unreleased version for client-only changes.
* Pointing to an unreleased version for server changes.
* Pointing to a fork for either of the above.

The gist of it is we'll now grab the version from `replace` lines too, and we just maintain a custom mapping from replaced versions to full image references, which can either be manually built ourselves or just pointed to the upstream `moby/buildkit:master` equivalent (but with a digest for stability).